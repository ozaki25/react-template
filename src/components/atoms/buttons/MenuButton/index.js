import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const MenuButton = ({ children, onClick }) => <Button onClick={onClick}>{children}</Button>;

export const Button = styled.button`
  border: solid 1px #eee;
  box-shadow: 1px 1px 1px #bdbdbd;
  color: #212121;
  display: inline-block;
  font-size: 14px;
  font-weight: bold;
  height: 44px;
  padding-left: 10px;
  padding-right: 10px;
  text-align: left;
  width: 300px;
  &:active {
    opacity: 0.5;
    box-shadow: 1px 1px 1px #f5f5f5;
  }
`;

MenuButton.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default MenuButton;
