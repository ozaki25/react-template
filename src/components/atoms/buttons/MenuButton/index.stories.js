import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import MenuButton from '.';

const stories = storiesOf('atoms/buttons', module);

stories.add(
  'MenuButton',
  withInfo('MenuButton')(() => (
    <MenuButton onClick={action('click button')}>{text('text', 'ボタン')}</MenuButton>
  )),
);
