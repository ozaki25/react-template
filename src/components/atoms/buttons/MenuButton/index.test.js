import React from 'react';
import { snapshot } from 'test/helpers';
import MenuButton from '.';

const props = {
  onClick: jest.fn(),
};

snapshot('MenuButton', <MenuButton {...props}>ボタン</MenuButton>);
