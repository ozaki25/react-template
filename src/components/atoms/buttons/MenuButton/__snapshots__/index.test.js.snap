// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: MenuButton 1`] = `
.c0 {
  border: solid 1px #eee;
  box-shadow: 1px 1px 1px #bdbdbd;
  color: #212121;
  display: inline-block;
  font-size: 14px;
  font-weight: bold;
  height: 44px;
  padding-left: 10px;
  padding-right: 10px;
  text-align: left;
  width: 300px;
}

.c0:active {
  opacity: 0.5;
  box-shadow: 1px 1px 1px #f5f5f5;
}

<MenuButton
  onClick={[Function]}
>
  <styled.button
    onClick={[Function]}
  >
    <button
      className="c0"
      onClick={[Function]}
    >
      ボタン
    </button>
  </styled.button>
</MenuButton>
`;
